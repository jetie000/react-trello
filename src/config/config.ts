export const config = {
  API_URL: process.env.REACT_APP_API_URL,
  BURGER_BREAKPOINT: 576
}
