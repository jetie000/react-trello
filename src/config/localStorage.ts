export const localStorageConfig = {
  TOKEN_LOCALSTORAGE: "token",
  THEME_LOCALSTORAGE: "page_theme",
  LANGUAGE_LOCALSTORAGE: "page_language",
  USERID_LOCALSTORAGE: "user_id"
}
