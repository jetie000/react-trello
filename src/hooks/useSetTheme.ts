

export const useSetTheme = () => {
  const setTheme = () => {
    document.documentElement.setAttribute(
      "data-bs-theme",
      localStorage.getItem(localStorage.THEME_LOCALSTORAGE) || "dark"
    )
  }
  return setTheme
}
