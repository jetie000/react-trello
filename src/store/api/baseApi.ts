import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
import { variables } from "@/variables";
import { api } from "@/http";

const axiosBaseQuery =
  ({ baseUrl } = { baseUrl: '' }) =>
    async ({ url, method, data, params, headers }: any) => {
      try {
        const result = await api({
          url: baseUrl + url,
          method,
          data,
          params,
          headers,
        })
        return { data: result.data }
      } catch (axiosError: any) {
        const err = axiosError
        return {
          error: {
            status: err.response?.status,
            data: err.response?.data || err.message,
          },
        }
      }
    }

export const baseApi = createApi({
  reducerPath: 'baseApi',
  baseQuery: axiosBaseQuery({
    baseUrl: variables.API_URL,
  }),
  tagTypes: ['User', 'Users'],
  endpoints: () => ({}),
});