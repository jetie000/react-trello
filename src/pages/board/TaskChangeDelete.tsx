import { useActions } from "@/hooks/useActions"
import { useChangeTaskMutation, useDeleteTaskMutation } from "@/store/api/task.api"
import { ITask } from "@/types/task.interface"
import { useEffect, useRef, useState } from "react"
import { Modal } from "bootstrap"
import UsersList from "@/components/usersList/UsersList"
import ModalWrapper from "@/components/modalWrapper/ModalWrapper"
import { useParams } from "react-router-dom"
import { RootState } from "@/store/store"
import { useSelector } from "react-redux"
import { variables } from "@/variables"

function TaskChangeDelete({ task }: { task: ITask | undefined }) {
  const { id } = useParams()
  const { language } = useSelector((state: RootState) => state.options)
  const { showToast } = useActions()
  const modalRefDelete = useRef<HTMLDivElement>(null)
  const changeTaskNameRef = useRef<HTMLInputElement>(null)
  const changeTaskDescRef = useRef<HTMLTextAreaElement>(null)
  const modalRefChange = useRef<HTMLDivElement>(null)
  const [
    changeTask,
    { isSuccess: isSuccessChange, isError: isErrorChange, isLoading: isLoadingChange }
  ] = useChangeTaskMutation()
  const [
    deleteTask,
    { isSuccess: isSuccessDelete, isError: isErrorDelete, isLoading: isLoadingDelete }
  ] = useDeleteTaskMutation()
  const [userIds, setUserIds] = useState<number[]>([])

  useEffect(() => {
    if (task?.users.length) {
      setUserIds(task.users.map(u => u.id))
    }
  }, [task?.users])

  const changeTaskClick = () => {
    if (
      changeTaskNameRef.current &&
      changeTaskDescRef.current &&
      changeTaskNameRef.current.value !== "" &&
      task
    ) {
      changeTask({
        id: task.id,
        name: changeTaskNameRef.current.value,
        description: changeTaskDescRef.current.value,
        userIds,
        columnId: task.columnId
      })
    }
  }

  const deleteTaskClick = () => {
    if (modalRefDelete.current) {
      const myModal = Modal.getOrCreateInstance("#" + modalRefDelete.current?.id)
      myModal.show()
    }
  }

  useEffect(() => {
    if (isSuccessChange) {
      showToast(variables.LANGUAGES[language].TASK_CHANGED)
    }
    if (isErrorChange) {
      showToast(variables.LANGUAGES[language].ERROR_REQUEST)
    }
  }, [isLoadingChange])

  useEffect(() => {
    if (isSuccessDelete) {
      if (modalRefDelete.current) {
        const myModal = Modal.getOrCreateInstance("#" + modalRefDelete.current?.id)
        myModal.hide()
      }
      if (modalRefChange.current) {
        const myModal = Modal.getOrCreateInstance("#" + modalRefChange.current?.id)
        myModal.hide()
      }
      showToast(variables.LANGUAGES[language].TASK_DELETED)
    }
    if (isErrorDelete) {
      showToast(variables.LANGUAGES[language].ERROR_REQUEST)
    }
  }, [isLoadingDelete])

  return (
    <ModalWrapper
      id="changeTask"
      title={variables.LANGUAGES[language].CHANGE_TASK}
      size="md"
      ref={modalRefChange}
    >
      <div className="d-flex flex-column">
        <label htmlFor="inputTaskNameChange">{variables.LANGUAGES[language].NAME}</label>
        <input
          className="form-control mb-2"
          id="inputTaskNameChange"
          placeholder={variables.LANGUAGES[language].ENTER_NAME}
          ref={changeTaskNameRef}
          defaultValue={task?.name}
        />
        <label htmlFor="inputTaskDescChange">{variables.LANGUAGES[language].DESCRIPTION}</label>
        <textarea
          className="form-control mb-2"
          id="inputTaskDescChange"
          placeholder={variables.LANGUAGES[language].ENTER_DESCRIPTION}
          ref={changeTaskDescRef}
          defaultValue={task?.description}
        />
        <UsersList userIds={userIds} setUserIds={setUserIds} boardId={Number(id)} />
        <button className="btn btn-primary mt-2 mb-2" onClick={changeTaskClick}>
          {variables.LANGUAGES[language].CHANGE_TASK}
        </button>
        <button className="btn btn-danger" onClick={deleteTaskClick}>
          {variables.LANGUAGES[language].DELETE_TASK}
        </button>
        <ModalWrapper
          id="deleteTask"
          title={variables.LANGUAGES[language].DELETE_TASK}
          size="sm"
          ref={modalRefDelete}
        >
          <div className="d-flex flex-column gap-2">
            <div>{variables.LANGUAGES[language].SURE_DELETE_TASK}</div>
            <button
              className="btn btn-danger"
              onClick={() => deleteTask({ taskId: task?.id || 0, boardId: Number(id) || 0 })}
            >
              {variables.LANGUAGES[language].DELETE_TASK}
            </button>
          </div>
        </ModalWrapper>
      </div>
    </ModalWrapper>
  )
}

export default TaskChangeDelete
