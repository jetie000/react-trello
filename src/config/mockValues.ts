import { IBoard } from "@/types/board.interface"
import { IColumn } from "@/types/column.interface"
import { IUserResponse } from "@/types/user.interface"

export const mockColumn: IColumn = {
  id: 1,
  boardId: 1,
  name: "TODO",
  order: 1,
  tasks: [
    {
      id: 1,
      columnId: 1,
      name: "Task 1",
      creationDate: new Date(0),
      moveDate: new Date(0),
      description: "Description of Task 1",
      users: [
        { id: 1, fullName: "User 1", email: "user1@gmail.com", loginDate: new Date() },
        { id: 2, fullName: "User 2", email: "user2@gmail.com", loginDate: new Date() },
        { id: 3, fullName: "User 3", email: "user3@gmail.com", loginDate: new Date() }
      ],
      Column: {
        id: 1,
        boardId: 1,
        name: "TODO",
        order: 1
      }
    },
    {
      id: 2,
      columnId: 1,
      name: "Task 2",
      creationDate: new Date(),
      moveDate: new Date(),
      description: "Description of Task 2",
      users: [
        { id: 4, fullName: "User 4", email: "user4@gmail.com", loginDate: new Date() },
        { id: 5, fullName: "User 5", email: "user5@gmail.com", loginDate: new Date() }
      ],
      Column: {
        id: 1,
        boardId: 1,
        name: "TODO",
        order: 1
      }
    }
  ]
}

export const mockUsers = [
  {
    id: 1,
    email: "1@gmail.com",
    fullName: "AA AA",
    loginDate: new Date()
  },
  {
    id: 2,
    email: "2@gmail.com",
    fullName: "BB BB",
    loginDate: new Date()
  },
  {
    id: 3,
    email: "3@gmail.com",
    fullName: "CC CC",
    loginDate: new Date()
  }
]
export const mockReturnUsers = (users: IUserResponse[]) => ({
  isLoading: false,
  isError: false,
  refetch: jest.fn(),
  data: users
})
export const mockBoards: IBoard[] = [
  {
    id: 1,
    creatorId: 1,
    name: "Board1",
    columns: [
      mockColumn,
      {
        id: 2,
        boardId: 1,
        name: "PENDING",
        order: 2
      },
      {
        id: 3,
        boardId: 1,
        name: "DONE",
        order: 3
      }
    ]
  },
  {
    id: 2,
    creatorId: 1,
    name: "Board2"
  },
  {
    id: 3,
    creatorId: 1,
    name: "Board3"
  }
]

export const mockReturnMutation = (isLoading: boolean, isPressed: boolean) => ({
  isLoading: isLoading && isPressed,
  isError: false,
  isSuccess: !isLoading && isPressed,
  refetch: jest.fn(),
  reset: jest.fn()
})
