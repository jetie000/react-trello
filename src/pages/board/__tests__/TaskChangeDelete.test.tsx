import * as RRD from "react-router-dom"
import * as reduxHooks from "react-redux"
import * as actions from "@/hooks/useActions"
import * as taskApi from "@/store/api/task.api"
import * as userApi from "@/store/api/user.api"
import { fireEvent, render, screen } from "@testing-library/react"
import { languages } from "@/config/languages"
import { skipToken } from "@reduxjs/toolkit/query"
import TaskChangeDelete from "../TaskChangeDelete"
import { mockColumn, mockReturnUsers, mockUsers } from "../../../config/mockValues"
import { mockReturnMutation } from "@/config/mockValues"

jest.mock("@/store/store")
jest.mock("react-redux")
jest.mock("react-router-dom")
jest.mock("@/hooks/useActions")
jest.mock("@/store/api/user.api")
jest.mock("@/store/api/task.api")
const mockUseSelector = jest.spyOn(reduxHooks, "useSelector")
const mockUseParams = jest.spyOn(RRD, "useParams")
const mockUseActions = jest.spyOn(actions, "useActions")
const mockChange = jest.spyOn(taskApi, "useChangeTaskMutation")
const mockDelete = jest.spyOn(taskApi, "useDeleteTaskMutation")
const mockGetByIds = jest.spyOn(userApi, "useGetByIdsQuery")
const mockSearchUsers = jest.spyOn(userApi, "useSearchUsersQuery")

const mockReturnSelectorValue = {
  language: 1,
  id: 1,
  token: "token"
}
const mockUseParamsReturn = { id: "1" }
const mockChangeFunc = jest.fn()
const mockDeleteFunc = jest.fn()
const showToastMock = jest.fn()
const mockReturnActionsValue = { showToast: showToastMock }

const nameTask = "Task 11"
const descriptionTask = "Desc Task 11"

describe("TaskChangeDelete", () => {
  beforeEach(() => {
    jest.restoreAllMocks()
    jest.resetAllMocks()
    mockGetByIds.mockImplementation((ids: number[] | typeof skipToken) => {
      if (typeof ids !== "symbol")
        return mockReturnUsers(mockUsers.filter(u => ids.some(id => id === u.id))) as any
    })
    mockSearchUsers.mockReturnValue(mockReturnUsers(mockUsers))
    mockUseSelector.mockReturnValue(mockReturnSelectorValue)
    mockUseActions.mockReturnValue(mockReturnActionsValue as any)
    mockChange.mockReturnValue([mockChangeFunc, mockReturnMutation(false, false)])
    mockDelete.mockReturnValue([mockDeleteFunc, mockReturnMutation(false, false)])
    mockUseParams.mockReturnValue(mockUseParamsReturn)
  })
  it("should render component with all fields and buttons", () => {
    const component = render(<TaskChangeDelete task={mockColumn.tasks![0]} />)
    expect(
      screen.getByPlaceholderText(languages[mockReturnSelectorValue.language].ENTER_NAME)
    ).toBeInTheDocument()
    expect(
      screen.getByPlaceholderText(languages[mockReturnSelectorValue.language].ENTER_DESCRIPTION)
    ).toBeInTheDocument()
    expect(screen.getByTestId("change-task-btn")).toBeInTheDocument()
    expect(component).toMatchSnapshot()
  })
  it("should show enter data message when provided invalid data when changing task", () => {
    const component = render(<TaskChangeDelete task={mockColumn.tasks![0]} />)
    fireEvent.change(
      screen.getByPlaceholderText(languages[mockReturnSelectorValue.language].ENTER_NAME),
      {
        target: { value: "" }
      }
    )
    fireEvent.click(screen.getByTestId("change-task-btn"))
    expect(showToastMock).toHaveBeenCalledWith(
      languages[mockReturnSelectorValue.language].INPUT_DATA
    )
    expect(showToastMock).toHaveBeenCalledTimes(1)
  })
  it("should change task when all fields are filled correctly", () => {
    const component = render(<TaskChangeDelete task={mockColumn.tasks![0]} />)
    expect(mockUseActions).toHaveBeenCalled()
    expect(mockChange).toHaveBeenCalled()
    expect(mockUseSelector).toHaveBeenCalled()
    fireEvent.change(
      screen.getByPlaceholderText(languages[mockReturnSelectorValue.language].ENTER_NAME),
      {
        target: { value: nameTask }
      }
    )
    fireEvent.change(
      screen.getByPlaceholderText(languages[mockReturnSelectorValue.language].ENTER_DESCRIPTION),
      {
        target: { value: descriptionTask }
      }
    )
    fireEvent.click(screen.getByTestId("change-task-btn"))
    mockChange.mockReturnValue([mockChangeFunc, mockReturnMutation(true, true)])
    component.rerender(<TaskChangeDelete task={mockColumn.tasks![0]} />)
    mockChange.mockReturnValue([mockChangeFunc, mockReturnMutation(false, true)])
    component.rerender(<TaskChangeDelete task={mockColumn.tasks![0]} />)
    expect(mockChangeFunc).toHaveBeenCalledWith({
      id: mockColumn.tasks![0].id,
      columnId: mockColumn.tasks![0].columnId,
      name: nameTask,
      description: descriptionTask,
      userIds: mockColumn.tasks![0].users.map(u => u.id)
    })
    expect(showToastMock).toHaveBeenCalledWith(
      languages[mockReturnSelectorValue.language].TASK_CHANGED
    )
    expect(showToastMock).toHaveBeenCalledTimes(1)
  })
  it("should delete task", () => {
    const component = render(<TaskChangeDelete task={mockColumn.tasks![0]} />)
    fireEvent.click(screen.getByTestId("delete-task-btn"))
    mockDelete.mockReturnValue([mockDeleteFunc, mockReturnMutation(true, true)])
    component.rerender(<TaskChangeDelete task={mockColumn.tasks![0]} />)
    mockDelete.mockReturnValue([mockDeleteFunc, mockReturnMutation(false, true)])
    component.rerender(<TaskChangeDelete task={mockColumn.tasks![0]} />)
    expect(mockDeleteFunc).toHaveBeenCalledWith({
      taskId: mockColumn.tasks![0].id,
      boardId: Number(mockUseParamsReturn.id)
    })
    expect(showToastMock).toHaveBeenCalledWith(
      languages[mockReturnSelectorValue.language].TASK_DELETED
    )
    expect(showToastMock).toHaveBeenCalledTimes(1)
  })
})
