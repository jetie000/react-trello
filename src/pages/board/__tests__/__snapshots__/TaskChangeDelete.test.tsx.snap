// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskChangeDelete should render component with all fields and buttons 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-hidden="true"
        aria-labelledby="changeTaskLabel"
        class="modal  fade"
        data-testid="changeTask"
        id="changeTask"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="modal-dialog modal-dialog-centered modal-md"
          role="document"
        >
          <div
            class="modal-content"
          >
            <div
              class="modal-header"
            >
              <h5
                class="modal-title"
                id="changeTaskLabel"
              >
                Change task
              </h5>
              <button
                aria-label="Close"
                class="btn-close"
                data-bs-dismiss="modal"
                type="button"
              />
            </div>
            <div
              class="modal-body"
            >
              <div
                class="d-flex flex-column"
              >
                <label
                  for="inputTaskNameChange"
                >
                  Name
                </label>
                <input
                  class="form-control mb-2"
                  id="inputTaskNameChange"
                  placeholder="Input name"
                  value="Task 1"
                />
                <label
                  for="inputTaskDescChange"
                >
                  Description (optional)
                </label>
                <textarea
                  class="form-control mb-2"
                  id="inputTaskDescChange"
                  placeholder="Enter description"
                >
                  Description of Task 1
                </textarea>
                <div
                  class="mb-1"
                >
                  <label>
                    Users
                  </label>
                  <div
                    class="input-group"
                  >
                    <input
                      class="form-control w-50"
                      data-testid="search-users-input"
                      placeholder="Enter request"
                      type="text"
                    />
                    <button
                      class="btn btn-secondary d-flex align-items-center add-board-behind-canvas"
                      type="button"
                    >
                      <svg
                        class="bi bi-x-lg"
                        fill="currentColor"
                        height="20"
                        viewBox="0 0 16 16"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M2.146 2.854a.5.5 0 1 1 .708-.708L8 7.293l5.146-5.147a.5.5 0 0 1 .708.708L8.707 8l5.147 5.146a.5.5 0 0 1-.708.708L8 8.707l-5.146 5.147a.5.5 0 0 1-.708-.708L7.293 8z"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
                <ul
                  class="list-group users-list"
                  data-testid="users-list"
                >
                  <li
                    class="list-group-item d-flex justify-content-between"
                  >
                    <div
                      class="d-flex flex-column"
                    >
                      <div>
                        1@gmail.com
                      </div>
                      <div>
                        AA AA
                      </div>
                    </div>
                    <button
                      class="btn btn-danger"
                      data-testid="remove-user-1"
                    >
                      <svg
                        class="bi bi-trash3-fill"
                        fill="currentColor"
                        height="16"
                        viewBox="0 0 16 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M11 1.5v1h3.5a.5.5 0 0 1 0 1h-.538l-.853 10.66A2 2 0 0 1 11.115 16h-6.23a2 2 0 0 1-1.994-1.84L2.038 3.5H1.5a.5.5 0 0 1 0-1H5v-1A1.5 1.5 0 0 1 6.5 0h3A1.5 1.5 0 0 1 11 1.5m-5 0v1h4v-1a.5.5 0 0 0-.5-.5h-3a.5.5 0 0 0-.5.5M4.5 5.029l.5 8.5a.5.5 0 1 0 .998-.06l-.5-8.5a.5.5 0 1 0-.998.06m6.53-.528a.5.5 0 0 0-.528.47l-.5 8.5a.5.5 0 0 0 .998.058l.5-8.5a.5.5 0 0 0-.47-.528M8 4.5a.5.5 0 0 0-.5.5v8.5a.5.5 0 0 0 1 0V5a.5.5 0 0 0-.5-.5"
                        />
                      </svg>
                    </button>
                  </li>
                  <li
                    class="list-group-item d-flex justify-content-between"
                  >
                    <div
                      class="d-flex flex-column"
                    >
                      <div>
                        2@gmail.com
                      </div>
                      <div>
                        BB BB
                      </div>
                    </div>
                    <button
                      class="btn btn-danger"
                      data-testid="remove-user-2"
                    >
                      <svg
                        class="bi bi-trash3-fill"
                        fill="currentColor"
                        height="16"
                        viewBox="0 0 16 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M11 1.5v1h3.5a.5.5 0 0 1 0 1h-.538l-.853 10.66A2 2 0 0 1 11.115 16h-6.23a2 2 0 0 1-1.994-1.84L2.038 3.5H1.5a.5.5 0 0 1 0-1H5v-1A1.5 1.5 0 0 1 6.5 0h3A1.5 1.5 0 0 1 11 1.5m-5 0v1h4v-1a.5.5 0 0 0-.5-.5h-3a.5.5 0 0 0-.5.5M4.5 5.029l.5 8.5a.5.5 0 1 0 .998-.06l-.5-8.5a.5.5 0 1 0-.998.06m6.53-.528a.5.5 0 0 0-.528.47l-.5 8.5a.5.5 0 0 0 .998.058l.5-8.5a.5.5 0 0 0-.47-.528M8 4.5a.5.5 0 0 0-.5.5v8.5a.5.5 0 0 0 1 0V5a.5.5 0 0 0-.5-.5"
                        />
                      </svg>
                    </button>
                  </li>
                  <li
                    class="list-group-item d-flex justify-content-between"
                  >
                    <div
                      class="d-flex flex-column"
                    >
                      <div>
                        3@gmail.com
                      </div>
                      <div>
                        CC CC
                      </div>
                    </div>
                    <button
                      class="btn btn-danger"
                      data-testid="remove-user-3"
                    >
                      <svg
                        class="bi bi-trash3-fill"
                        fill="currentColor"
                        height="16"
                        viewBox="0 0 16 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M11 1.5v1h3.5a.5.5 0 0 1 0 1h-.538l-.853 10.66A2 2 0 0 1 11.115 16h-6.23a2 2 0 0 1-1.994-1.84L2.038 3.5H1.5a.5.5 0 0 1 0-1H5v-1A1.5 1.5 0 0 1 6.5 0h3A1.5 1.5 0 0 1 11 1.5m-5 0v1h4v-1a.5.5 0 0 0-.5-.5h-3a.5.5 0 0 0-.5.5M4.5 5.029l.5 8.5a.5.5 0 1 0 .998-.06l-.5-8.5a.5.5 0 1 0-.998.06m6.53-.528a.5.5 0 0 0-.528.47l-.5 8.5a.5.5 0 0 0 .998.058l.5-8.5a.5.5 0 0 0-.47-.528M8 4.5a.5.5 0 0 0-.5.5v8.5a.5.5 0 0 0 1 0V5a.5.5 0 0 0-.5-.5"
                        />
                      </svg>
                    </button>
                  </li>
                </ul>
                <button
                  class="btn btn-primary mt-2 mb-2"
                  data-testid="change-task-btn"
                >
                  Change task
                </button>
                <button
                  class="btn btn-danger"
                >
                  Delete task
                </button>
                <div
                  aria-hidden="true"
                  aria-labelledby="deleteTaskLabel"
                  class="modal  fade"
                  data-testid="deleteTask"
                  id="deleteTask"
                  role="dialog"
                  tabindex="-1"
                >
                  <div
                    class="modal-dialog modal-dialog-centered modal-sm"
                    role="document"
                  >
                    <div
                      class="modal-content"
                    >
                      <div
                        class="modal-header"
                      >
                        <h5
                          class="modal-title"
                          id="deleteTaskLabel"
                        >
                          Delete task
                        </h5>
                        <button
                          aria-label="Close"
                          class="btn-close"
                          data-bs-dismiss="modal"
                          type="button"
                        />
                      </div>
                      <div
                        class="modal-body"
                      >
                        <div
                          class="d-flex flex-column gap-2"
                        >
                          <div>
                            Are you sure you want delete this task?
                          </div>
                          <button
                            class="btn btn-danger"
                            data-testid="delete-task-btn"
                          >
                            Delete task
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-hidden="true"
      aria-labelledby="changeTaskLabel"
      class="modal  fade"
      data-testid="changeTask"
      id="changeTask"
      role="dialog"
      tabindex="-1"
    >
      <div
        class="modal-dialog modal-dialog-centered modal-md"
        role="document"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <h5
              class="modal-title"
              id="changeTaskLabel"
            >
              Change task
            </h5>
            <button
              aria-label="Close"
              class="btn-close"
              data-bs-dismiss="modal"
              type="button"
            />
          </div>
          <div
            class="modal-body"
          >
            <div
              class="d-flex flex-column"
            >
              <label
                for="inputTaskNameChange"
              >
                Name
              </label>
              <input
                class="form-control mb-2"
                id="inputTaskNameChange"
                placeholder="Input name"
                value="Task 1"
              />
              <label
                for="inputTaskDescChange"
              >
                Description (optional)
              </label>
              <textarea
                class="form-control mb-2"
                id="inputTaskDescChange"
                placeholder="Enter description"
              >
                Description of Task 1
              </textarea>
              <div
                class="mb-1"
              >
                <label>
                  Users
                </label>
                <div
                  class="input-group"
                >
                  <input
                    class="form-control w-50"
                    data-testid="search-users-input"
                    placeholder="Enter request"
                    type="text"
                  />
                  <button
                    class="btn btn-secondary d-flex align-items-center add-board-behind-canvas"
                    type="button"
                  >
                    <svg
                      class="bi bi-x-lg"
                      fill="currentColor"
                      height="20"
                      viewBox="0 0 16 16"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M2.146 2.854a.5.5 0 1 1 .708-.708L8 7.293l5.146-5.147a.5.5 0 0 1 .708.708L8.707 8l5.147 5.146a.5.5 0 0 1-.708.708L8 8.707l-5.146 5.147a.5.5 0 0 1-.708-.708L7.293 8z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
              <ul
                class="list-group users-list"
                data-testid="users-list"
              >
                <li
                  class="list-group-item d-flex justify-content-between"
                >
                  <div
                    class="d-flex flex-column"
                  >
                    <div>
                      1@gmail.com
                    </div>
                    <div>
                      AA AA
                    </div>
                  </div>
                  <button
                    class="btn btn-danger"
                    data-testid="remove-user-1"
                  >
                    <svg
                      class="bi bi-trash3-fill"
                      fill="currentColor"
                      height="16"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M11 1.5v1h3.5a.5.5 0 0 1 0 1h-.538l-.853 10.66A2 2 0 0 1 11.115 16h-6.23a2 2 0 0 1-1.994-1.84L2.038 3.5H1.5a.5.5 0 0 1 0-1H5v-1A1.5 1.5 0 0 1 6.5 0h3A1.5 1.5 0 0 1 11 1.5m-5 0v1h4v-1a.5.5 0 0 0-.5-.5h-3a.5.5 0 0 0-.5.5M4.5 5.029l.5 8.5a.5.5 0 1 0 .998-.06l-.5-8.5a.5.5 0 1 0-.998.06m6.53-.528a.5.5 0 0 0-.528.47l-.5 8.5a.5.5 0 0 0 .998.058l.5-8.5a.5.5 0 0 0-.47-.528M8 4.5a.5.5 0 0 0-.5.5v8.5a.5.5 0 0 0 1 0V5a.5.5 0 0 0-.5-.5"
                      />
                    </svg>
                  </button>
                </li>
                <li
                  class="list-group-item d-flex justify-content-between"
                >
                  <div
                    class="d-flex flex-column"
                  >
                    <div>
                      2@gmail.com
                    </div>
                    <div>
                      BB BB
                    </div>
                  </div>
                  <button
                    class="btn btn-danger"
                    data-testid="remove-user-2"
                  >
                    <svg
                      class="bi bi-trash3-fill"
                      fill="currentColor"
                      height="16"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M11 1.5v1h3.5a.5.5 0 0 1 0 1h-.538l-.853 10.66A2 2 0 0 1 11.115 16h-6.23a2 2 0 0 1-1.994-1.84L2.038 3.5H1.5a.5.5 0 0 1 0-1H5v-1A1.5 1.5 0 0 1 6.5 0h3A1.5 1.5 0 0 1 11 1.5m-5 0v1h4v-1a.5.5 0 0 0-.5-.5h-3a.5.5 0 0 0-.5.5M4.5 5.029l.5 8.5a.5.5 0 1 0 .998-.06l-.5-8.5a.5.5 0 1 0-.998.06m6.53-.528a.5.5 0 0 0-.528.47l-.5 8.5a.5.5 0 0 0 .998.058l.5-8.5a.5.5 0 0 0-.47-.528M8 4.5a.5.5 0 0 0-.5.5v8.5a.5.5 0 0 0 1 0V5a.5.5 0 0 0-.5-.5"
                      />
                    </svg>
                  </button>
                </li>
                <li
                  class="list-group-item d-flex justify-content-between"
                >
                  <div
                    class="d-flex flex-column"
                  >
                    <div>
                      3@gmail.com
                    </div>
                    <div>
                      CC CC
                    </div>
                  </div>
                  <button
                    class="btn btn-danger"
                    data-testid="remove-user-3"
                  >
                    <svg
                      class="bi bi-trash3-fill"
                      fill="currentColor"
                      height="16"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M11 1.5v1h3.5a.5.5 0 0 1 0 1h-.538l-.853 10.66A2 2 0 0 1 11.115 16h-6.23a2 2 0 0 1-1.994-1.84L2.038 3.5H1.5a.5.5 0 0 1 0-1H5v-1A1.5 1.5 0 0 1 6.5 0h3A1.5 1.5 0 0 1 11 1.5m-5 0v1h4v-1a.5.5 0 0 0-.5-.5h-3a.5.5 0 0 0-.5.5M4.5 5.029l.5 8.5a.5.5 0 1 0 .998-.06l-.5-8.5a.5.5 0 1 0-.998.06m6.53-.528a.5.5 0 0 0-.528.47l-.5 8.5a.5.5 0 0 0 .998.058l.5-8.5a.5.5 0 0 0-.47-.528M8 4.5a.5.5 0 0 0-.5.5v8.5a.5.5 0 0 0 1 0V5a.5.5 0 0 0-.5-.5"
                      />
                    </svg>
                  </button>
                </li>
              </ul>
              <button
                class="btn btn-primary mt-2 mb-2"
                data-testid="change-task-btn"
              >
                Change task
              </button>
              <button
                class="btn btn-danger"
              >
                Delete task
              </button>
              <div
                aria-hidden="true"
                aria-labelledby="deleteTaskLabel"
                class="modal  fade"
                data-testid="deleteTask"
                id="deleteTask"
                role="dialog"
                tabindex="-1"
              >
                <div
                  class="modal-dialog modal-dialog-centered modal-sm"
                  role="document"
                >
                  <div
                    class="modal-content"
                  >
                    <div
                      class="modal-header"
                    >
                      <h5
                        class="modal-title"
                        id="deleteTaskLabel"
                      >
                        Delete task
                      </h5>
                      <button
                        aria-label="Close"
                        class="btn-close"
                        data-bs-dismiss="modal"
                        type="button"
                      />
                    </div>
                    <div
                      class="modal-body"
                    >
                      <div
                        class="d-flex flex-column gap-2"
                      >
                        <div>
                          Are you sure you want delete this task?
                        </div>
                        <button
                          class="btn btn-danger"
                          data-testid="delete-task-btn"
                        >
                          Delete task
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
